velox/functions/prestosql/BinaryFunctions.h

```diff
@@ -436,4 +436,107 @@ struct ToIEEE754Bits32 {
   }
 };
 
+template <bool lpad, bool isAscii, typename TOut, typename TIn>
+FOLLY_ALWAYS_INLINE void
+pad(TOut& output, const TIn& input, const int64_t size, const TIn& padding) {
+  static constexpr size_t padMaxSize = 1024 * 1024; // 1MB
+
+  VELOX_USER_CHECK(
+      size >= 0 && size <= padMaxSize,
+      "pad size must be in the range [0..{})",
+      padMaxSize);
+  VELOX_USER_CHECK(padding.size() > 0, "padding bytes must not be empty");
+
+  int64_t inputSize = input.size();
+  // Input has at most size length, truncate it if necessary
+  // and return it as the result.
+  if (UNLIKELY(inputSize >= size)) {
+    output.resize(size);
+    std::memcpy(output.data(), input.data(), size);
+    return;
+  }
+
+  int64_t paddingSize = padding.size();
+  // How much length do we need to add to input.
+  int64_t fullPaddingLength = size - inputSize;
+  // How many full copies of padding need to be added.
+  int64_t fullPadCopies = fullPaddingLength / paddingSize;
+  // If the length of padding does not evenly divide the length of the
+  // padding we need to add, how long of a prefix of padding needs to be
+  // added at the end of the padding.  Will be 0 if it is evenly divisible.
+  size_t padPrefixByteLength = fullPaddingLength % paddingSize;
+  int64_t fullPaddingByteLength =
+      padding.size() * fullPadCopies + padPrefixByteLength;
+  // The final size of the output string in bytes.
+  int64_t outputByteLength = input.size() + fullPaddingByteLength;
+  // What byte index in the output to start writing the padding at.
+  int64_t paddingOffset;
+
+  output.resize(outputByteLength);
+
+  if constexpr (lpad) {
+    paddingOffset = 0;
+    // Copy input after the padding.
+    std::memcpy(
+        output.data() + fullPaddingByteLength, input.data(), input.size());
+  } else {
+    paddingOffset = input.size();
+    // Copy input at the beginning of the output.
+    std::memcpy(output.data(), input.data(), input.size());
+  }
+
+  for (int i = 0; i < fullPadCopies; i++) {
+    std::memcpy(
+        output.data() + paddingOffset + i * padding.size(),
+        padding.data(),
+        padding.size());
+  }
+
+  // Add the final prefix of padding to the padding in output (if necessary).
+  std::memcpy(
+      output.data() + paddingOffset + fullPadCopies * padding.size(),
+      padding.data(),
+      padPrefixByteLength);
+}
+
+/// Pad functions
+/// lpad(binary, size, padbinary) -> varbinary
```

Comment:
Why isn't there an lpad for varchar ?

Reply:
there's a function:: lpad(string, size, padstring) -> varchar (also rpad) in string.rst which has already been implemented